azurerm/internal/services/web/resource_arm_function_app.go

```diff
@@ -870,3 +945,24 @@ func flattenFunctionAppSiteCredential(input *web.UserProperties) []interface{} {
 
 	return append(results, result)
 }
+
+func flattenFunctionAppIpRestriction(input *[]web.IPSecurityRestriction) []interface{} {
+	restrictions := make([]interface{}, 0)
+
+	if input == nil {
+		return restrictions
+	}
+
+	for _, v := range *input {
+		block := make(map[string]interface{})
+		if ip := v.IPAddress; ip != nil {
+			block["ip_address"] = *ip
+		}
```

Comment:
we need to ensure there's a value set for this nested field even if it's unset - as such can we make this:

```suggestion
			ipAddress := ""
			if v.IPAddress != nil {
				ipAddress = *v.IPAddress
			}
			block["ip_address"] = ipAddress
```